### PR TITLE
Intubations checklist screen

### LINKED ISSUE #38: Build out the intubations checklist screen
From the main menu there is a button which takes you to the intubation checklist screen.  That screen is currently **really** basic.

**We need to build build out a detailed version of the intubation checklist screen**

@greggmiller Can you please provide some information in this issue on what should be in this section?  Once that is done assign the issue to Marc Edwards.

@marcedwards Once greg has provided info, can you provide some design guidance in this issue in on how it should look.  Once this is done, can you label this issue as "Ready for Dev" so that one of the devs can pick it up.

### PR DESCRIPTION
**Pull request by ericjiang97**
This PR builds out the intubations checklist screen

- [x] `release-notes.txt` has been updated.

## Changes

- Initial work on the Checklist ✅ 
- Added reusable checkbox component 🔧 
- Added more models for future proofing.

## Screenshots

https://ejiang.d.pr/piupi4

## Things to note

### CODE REVIEW

File: lib/view/intubation/checklist/intubation_checklist_page.dart [hunk 1]
```diff
@@ -1,28 +1,48 @@
...
+          iconTheme: AppStyles.appBarIconTheme,
+          backgroundColor: AppColors.green50,
+          title: const Text(
+            'Intubation Checklist',
+            style: AppStyles.textSemiBold,
+          ),
+          bottom: TabBar(
+            unselectedLabelColor: AppColors.grey600,
+            indicatorColor: AppColors.blackAlpha800,
+            labelColor: AppColors.blackAlpha800,
```
Comment (rockgecko-development) [inline lib/view/intubation/checklist/intubation_checklist_page.dart]: The tab bar indicator style should be bold according to mocks. I used the following to style the tab bar in intubation guidance #133:
```
labelColor: AppColors.blackAlpha800,
labelStyle: AppStyles.textSemiBold,
unselectedLabelColor: AppColors.tabBarDeselectedText,
indicatorColor: AppColors.green900,
```
https://github.com/Western-Health-Covid19-Collaboration/wh_covid19_app/blob/464be6825dab06fcb3a259ffb256f78aea9fc9cc/lib/view/intubation/guidance/intubation_guidance_page.dart#L21

**Review by rockgecko-development — APPROVED** [review]
[duplicate msg]

File: lib/view/intubation/checklist/intubation_checklist_page.dart [hunk 2]
```diff
@@ -1,28 +1,48 @@
...
+      child: Scaffold(
+        appBar: AppBar(
+          iconTheme: AppStyles.appBarIconTheme,
+          backgroundColor: AppColors.green50,
+          title: const Text(
+            'Intubation Checklist',
+            style: AppStyles.textSemiBold,
+          ),
+          bottom: TabBar(
+            unselectedLabelColor: AppColors.grey600,
```
Comment (rockgecko-development) [inline lib/view/intubation/checklist/intubation_checklist_page.dart]: The tab bar indicator style should be bold according to mocks. I used the following to style the tab bar in intubation guidance #133:
```
labelColor: AppColors.blackAlpha800,
labelStyle: AppStyles.textSemiBold,
unselectedLabelColor: AppColors.tabBarDeselectedText,
indicatorColor: AppColors.green900,
```
https://github.com/Western-Health-Covid19-Collaboration/wh_covid19_app/blob/464be6825dab06fcb3a259ffb256f78aea9fc9cc/lib/view/intubation/guidance/intubation_guidance_page.dart#L21

**Review by rockgecko-development — CHANGES_REQUESTED** [review]
Looks great! One minor thing re tab bar design

File: lib/hard_data.dart [hunk 3]
```diff
@@ -238,6 +239,77 @@ const List<PPEStepInfo> ppeOffMethod2Steps = [
...
+    ),
+  ]),
+  IntubationChecklist(title: 'Patient', checklist: [
+    IntubationChecklistItem(
+      title: 'ECG, BP, Sats',
+    ),
+    IntubationChecklistItem(
+        title: 'Pre-oxygenation', notes: ['FIO2 100%', 'Sitting position 45°']),
+    IntubationChecklistItem(
+      title: 'IV access x 2',
```
Comment (kimngyn) [inline lib/hard_data.dart]:
```suggestion
      title: 'IV access x 2', notes: ['1L fluid on pump set']
```

File: lib/hard_data.dart [hunk 4]
```diff
@@ -238,6 +239,77 @@ const List<PPEStepInfo> ppeOffMethod2Steps = [
...
+    IntubationChecklistItem(title: 'Supraglottic airway sized to pt'),
+    IntubationChecklistItem(title: 'Scalpel + bougie CICO kit'),
+    IntubationChecklistItem(title: 'Airway trolley/bronchoscope outside room'),
+    IntubationChecklistItem(title: 'ETCO2'),
+    IntubationChecklistItem(title: 'Viral filter'),
+  ]),
+  IntubationChecklist(title: 'Final', checklist: [
+    IntubationChecklistItem(title: 'Patient position optimal'),
+    IntubationChecklistItem(
+      title: 'Fluid runs easily',
```
Comment (kimngyn) [inline lib/hard_data.dart:296]:
```suggestion
      title: 'Fluid runs easily'
```

**Comment by rockgecko-development** [comment]
[deleted] my bad :blush:

**Review by rockgecko-development — APPROVED** [review]

**Pull request merged**